src/utils.ts

```diff
@@ -222,3 +222,16 @@ export function getGlobalCSSVariable(name: string) {
     `--${name}`,
   );
 }
+
+export const isServer = () => typeof window === "undefined";
+
+export const getPixelRatio = (): number =>
+  isServer() ? 1 : window.devicePixelRatio;
+
+export function createCanvasObject() {
+  if (isServer()) {
```

Comment:
Can you do feature checking, if document.createElement exists, then use it, otherwise use require("canvas")